src/main/groovy/life/qbic/ChecksumWriter.groovy

```diff
@@ -0,0 +1,9 @@
+package life.qbic
+
+interface ChecksumWriter {
+
+    def writeMatchingChecksum(String expectedChecksum, String computedChecksum, URL fileLocation)
+
+    def writeFailedChecksum(String expectedChecksum, String computedChecksum, URL fileLocation)
```

Comment:
matching and failing are not opposites.
successful and failing or matching and mismatching/conflicting/different